### ISSUE TITLE
Fatal error caused by emacs interlocking link

### ISSUE THREAD

**Issue by rogerz** [gruntjs/grunt-contrib-watch#239]
The watch task encounters a Fatal error when a temporary link is created by emacs.

```
Running "watch" task
Waiting...Fatal error: ENOENT, no such file or directory '/Users/zhangyujun/Workspace/xx/.#index.js'
```

The file is created by `emacs` when there is unsaved modification. It seems to be a special link which can not be read. See [emacs manual](http://www.gnu.org/software/emacs/manual/html_node/emacs/Interlocking.html#Interlocking)

```bash
ls -al
192:xx zhangyujun$ ls -al
total 24
drwxr-xr-x   8 zhangyujun  staff  272 11  8 21:02 .
lrwxr-xr-x   1 zhangyujun  staff   29 11  8 21:02 .#index.js -> zhangyujun@192.168.0.100.2274
drwxr-xr-x  17 zhangyujun  staff  578 11  8 20:52 ..
-rw-r--r--   1 zhangyujun  staff  208 11  8 20:58 Gruntfile.js

192:xx zhangyujun$ cat .#index.js
cat: .#index.js: No such file or directory
```

Is there anyway I can get rid of this error? I tried to negate it but it doesn't work

```
files: [
  '*.js',
  '!.#*'
]
```

Version

* grunt v0.4.1
* grunt-cli v0.1.9
* grunt-contrib-watch@0.5.3

**Comment by bzalasky** [comment]
If anyone else hits this issue, you can take an alternative approach and [keep your emacs backup and autosave files out of the way](http://emacsredux.com/blog/2013/05/09/keep-backup-and-auto-save-files-out-of-the-way/)...

**Comment by jahnreektor** [comment]
@jacksonrayhamilton's fork worked for me!

**Comment by jacksonrayhamilton** [comment]
Sadly `0.5.1` is still tagged as the latest release on the repository. Since a lot of my dependencies use gaze, and they fail whenever they encounter symlinks, I've created a program that automatically fixes *all* of my gaze dependencies. Here's my newest recommended fix:

```
npm install --save-dev fix-gaze
```

Add the following to your package.json:

```json
{
  "scripts": {
    "postinstall": "fix-gaze"
  }
}
```

**Comment by carljm** [comment]
@jacksonrayhamilton Thanks for that (truly horrendous) fix - it works great! :-) Can't wait for everything to stop using gaze...

**Comment by jacksonrayhamilton** [comment]
@carljm In retrospect I should have invested that time into moving the gaze package towards another (published) release. Unfortunately I am not very familiar with filesystems.

**Comment by shama** [comment]
I have backported and published `0.5.2` with the fix. Clearing the cache and re-installing should pick up the new version. Thanks!
